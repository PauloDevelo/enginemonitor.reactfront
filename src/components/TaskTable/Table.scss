/* Inject styles */
@import '~react-table-factory/styles/withFixedHeader';
@import '~react-table-factory/styles/withHeaderCellOverflow';
@import '~react-table-factory/styles/withSortingContext';
@import '~react-table-factory/styles/withInlineDetailsContext';
@import '~react-table-factory/styles/table';

/* Inject adaptive behavior */
.adaptive-col-name {
    display: none;
}

@media all and (max-width: 940px) {
    @import '~react-table-factory/styles/withAdaptive';

    table {
        @extend .adaptive-table;
    }

    .scrollable-table {
        @extend .adaptive-scrollable-table;
    }

    .scrollable-table {
        .multi-line {
            display: inline-flex;
            justify-content: flex-end;
            
            div + div::before  {
                content: ',';
                margin-right: 5px;
            }
        }
    }
}

/* Add effects (not necessary) */
.selectable-row {
    perspective: 40em;
    transform: rotateX(0);
    // transition: color .1s linear;
}

@media all and (min-width: 941px) {
    .selectable-row:not(.selected-row):hover,
    .selected-row
    {
        animation: rotate .3s ease-out;
    }

    @keyframes rotate {
        0% {
            transform: rotateX(30deg);
        }
        50% {
            transform: rotateX(60deg);
        }
        100% {
            transform: rotateX(0);
        }
    }
}

.selectable-row:not(.selected-row):hover,
.selected-row
{
    animation: rotate .3s ease-out;
    color: rgb(52, 121, 241);
}

.scrollable-table {
    height: calc(100% - 39px);
}

table.default-theme {
    --bg-color: #fff;
    --header-ft-size: 16px;
    --cell-ft-size: 14px;
    --v-align: left;
    --cell-padding: 0px 0px 0px 0px;
    --border-color: #e7e9eb;
    --even-bg-color: #f9f9f9;
    --cell-height: 32px;
}

.scrollable-table > .scrollable-table-wrapper, .scrollable-table > .scrollable-table-content > .scrollable-table-wrapper {
    padding: 0px 2px;
}