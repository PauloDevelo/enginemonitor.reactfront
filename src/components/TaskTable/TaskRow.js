import React from 'react';
import PropTypes from 'prop-types';
import { getContext, getTodoText, shorten } from '../../helpers/TaskHelper'; 

const getTrContext = (level) => "table-" + getContext(level)

const TaskRow = ({task, trStyle, onClick}) => {
	var shortenDescription = shorten(task.description.replace(/\n/g,'<br />'));
	var todoText = getTodoText(task)		
	var classNames = getTrContext(task.level) + " small";
		
	return(
        <tr style={trStyle} className={classNames} onClick={() => {
            if(onClick)
                onClick();
        }}>
			<td>{task.name}</td>
			<td>{todoText}</td>
			<td><div dangerouslySetInnerHTML={{ __html: shortenDescription }} /></td>
		</tr>
	);
}

TaskRow.propTypes = {
	task: PropTypes.object.isRequired,
	trStyle: PropTypes.object,
	onClick: PropTypes.func
};

export default TaskRow;