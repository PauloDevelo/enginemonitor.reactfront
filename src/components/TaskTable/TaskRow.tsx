import React, { CSSProperties } from 'react';
import PropTypes from 'prop-types';
import { getContext, getTodoText, shorten } from '../../helpers/TaskHelper'; 
import { Task } from '../../types/Types';

const getTrContext = (level: number) => "table-" + getContext(level);

type Props = {
	task: Task,
	trStyle: CSSProperties | undefined,
	onClick?: () => void
}

const TaskRow = ({task, trStyle, onClick}: Props) => {
	var shortenDescription = shorten(task.description.replace(/\n/g,'<br />'));
	var todoText = getTodoText(task)		
	var classNames = getTrContext(task.level) + " small";
		
	return(
        <tr style={trStyle} className={classNames} onClick={() => {
            if(onClick)
                onClick();
        }}>
			<td>{task.name}</td>
			<td>{todoText}</td>
			<td><div dangerouslySetInnerHTML={{ __html: shortenDescription }} /></td>
		</tr>
	);
}

TaskRow.propTypes = {
	task: PropTypes.object.isRequired,
	trStyle: PropTypes.object,
	onClick: PropTypes.func
};

export default TaskRow;