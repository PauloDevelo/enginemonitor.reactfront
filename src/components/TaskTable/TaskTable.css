.tasktable{
    display: flex;
    flex-direction: column;
}