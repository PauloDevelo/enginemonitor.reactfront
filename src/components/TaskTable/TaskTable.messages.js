import {defineMessages} from "react-intl";
import {type Messages} from "./MessageDescriptor"

const messages: Messages = {
	tasklistTitle: {
    id: "tasklistTitle",
    defaultMessage: "Task list",
		description: "Panel title"
  },
	createbutton: {
    id: "createbutton",
    defaultMessage: "Create",
		description: "Create button"
  },
	taskname: {
    id: "taskname",
    defaultMessage: "Task name",
		description: "Task name header"
  },
	todo: {
    id: "todo",
    defaultMessage: "To do",
		description: "To do header"
  },
	taskdesc: {
    id: "taskdesc",
    defaultMessage: "Description",
		description: "Description header"
  },
	shouldhavebeendone: {
    id: "shouldhavebeendone",
    defaultMessage: "Should have been done the ",
		description: "Sentence"
  },
	shouldbedone: {
    id: "shouldbedone",
    defaultMessage: "Should be done the ",
		description: "Sentence"
  },
	shouldhavebeendonein1: {
    id: "shouldhavebeendonein",
    defaultMessage: "Should have been done ",
		description: "Sentence"
  },
	shouldbedonein1: {
    id: "shouldbedonein1",
    defaultMessage: "Should be done in ",
		description: "Sentence"
  },
	shouldhavebeendonein2: {
    id: "shouldhavebeendonein2",
    defaultMessage: " equipment ago or ",
		description: "Sentence"
  },
	shouldbedonein2: {
    id: "shouldbedonein2",
    defaultMessage: " equipment or ",
		description: "Sentence"
  },
  tobedonemonth: {
    id: "tobedonemonth",
    defaultMessage: "To do every ",
		description: "Sentence"
  },
  orevery: {
    id: "orevery",
    defaultMessage: " or every ",
		description: "Sentence"
  },
  monthperiod: {
    id: "monthperiod",  
    defaultMessage: "{month, plural, one {month} other {# months}}",
    description: "Month period"
  },
  ackDate: {
    id: "ackDate",
    defaultMessage: "Realisation date",
    description: "Colum header"
  },
  remarks: {
    id: "remarks",
    defaultMessage: "Remarks",
    description: "Colum header"
  },
  age: {
		id: "age",
    defaultMessage: "Equipment Age",
		description: "Label"
	}
};

const definedMessages: typeof messages = defineMessages(messages);

export default definedMessages;