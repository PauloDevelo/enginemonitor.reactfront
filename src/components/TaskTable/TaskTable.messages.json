{
  "ackDate": {
    "defaultMessage": "Realisation date",
    "description": "Colum header",
    "id": "ackDate"
  },
  "age": {
    "defaultMessage": "Equipment Age",
    "description": "Label",
    "id": "age"
  },
  "createbutton": {
    "defaultMessage": "Create",
    "description": "Create button",
    "id": "createbutton"
  },
  "errorFetching": {
    "defaultMessage": "Something went wrong ...",
    "description": "Error message",
    "id": "errorFetching"
  },
  "remarks": {
    "defaultMessage": "Remarks",
    "description": "Colum header",
    "id": "remarks"
  },
  "status": {
    "defaultMessage": "Status",
    "description": "Label",
    "id": "status"
  },
  "taskdesc": {
    "defaultMessage": "Description",
    "description": "Description header",
    "id": "taskdesc"
  },
  "tasklistTitle": {
    "defaultMessage": "Task list",
    "description": "Panel title",
    "id": "tasklistTitle"
  },
  "taskname": {
    "defaultMessage": "Task name",
    "description": "Task name header",
    "id": "taskname"
  },
  
  "todo": {
    "defaultMessage": "To do",
    "description": "To do header",
    "id": "todo"
  }
}
