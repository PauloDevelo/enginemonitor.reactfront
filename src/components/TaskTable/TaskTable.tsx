import React, {Fragment, useEffect, useState, useCallback} from 'react';
import { Button, Badge } from 'reactstrap';
import { 
  composeDecorators,
  withInMemorySortingContext,
  withFixedHeader,
  withHeaderControl,
} from 'react-table-factory';

import { FormattedMessage, defineMessages } from 'react-intl';
import { faTasks, faPlusSquare } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import ModalEditTask from '../ModalEditTask/ModalEditTask';
import Loading from '../Loading/Loading';

import { getContext, getTodoText, shorten, getTodoValue, TaskTodo, createDefaultTask, getBadgeText } from '../../helpers/TaskHelper'; 
import { useEditModal } from '../../hooks/EditModalHook';

import { EquipmentModel, TaskModel } from '../../types/Types';

import '../../style/Table.scss';
import './TaskTable.css';

import jsonMessages from "./TaskTable.messages.json";
const taskTableMsg = defineMessages(jsonMessages);

type Props = {
	equipment?: EquipmentModel, 
	tasks: TaskModel[], 
	areTasksLoading: boolean, 
	onTaskSaved: (task: TaskModel) => void, 
	changeCurrentTask: (task: TaskModel) => void, 
	classNames?: string
};

type DisplayableTask = {
	task: TaskModel,
    name: string,
    todo: TaskTodo,
    shortenDescription: string,
	level: number,
	initialIndex: number
};

const Table = composeDecorators(
  withHeaderControl,
  withInMemorySortingContext(),
  withFixedHeader // should be last
)();

export const TaskTable = ({equipment, tasks, areTasksLoading, onTaskSaved, changeCurrentTask, classNames}: Props) => {
	classNames = classNames === undefined ? 'tasktable' : classNames + ' tasktable';
	const modalHook = useEditModal<TaskModel | undefined>(undefined);
	
	const getTasksData = useCallback(() => {
		let tasksData: DisplayableTask[] = [];
		if(tasks && equipment){
			tasksData = tasks.map((task, index) => {
				return {
					task: task,
					name: task.name,
					todo: getTodoValue(equipment, task),
					shortenDescription: shorten(task.description),
					level: task.level,
					initialIndex: index
				};
			});
		}

		return tasksData;
	}, [tasks, equipment]);

	const innerTaskCell = useCallback((task:DisplayableTask, content: JSX.Element, classNames?: string) => {
		classNames = classNames === undefined ? '' : classNames;
		classNames += ' table-' + getContext(task.level);

		return (
			<div onClick={() => changeCurrentTask(task.task)} className={classNames + ' innerTd clickable'} >
				{content}
			</div>
		);
	}, [changeCurrentTask]);

	const [columns, setColumns] = useState([
		{
			name: 'name',
			header: () => (
				<div className={'innerTdHead'}>
					<FormattedMessage {...taskTableMsg.taskname} />
				</div>
			),
			cell: (content: any) => {
				return innerTaskCell(content.data, <span>{content.data.name}</span>);
			},
			sortable: true
		},
		{
			name: 'initialIndex',
			header: () => (
				<div className={'innerTdHead'}>
					<FormattedMessage {...taskTableMsg.status} />
				</div>
			),
			cell: (content: any) => {
				return innerTaskCell(content.data, <Badge color={getContext(content.data.level)} pill className={'mx-auto my-auto'}>{getBadgeText(content.data.level)} </Badge>, 'd-flex');
			},
			sortable: true
		},
		{
			name: 'todo',
			header: () => (
				<div className={'text-center innerTdHead'}>
					<FormattedMessage {...taskTableMsg.todo}/>
				</div>
			),
			cell: (content: any) => {
				return innerTaskCell(content.data, getTodoText(content.data.todo));
			},
			sortable: false
		}
	]);

	useEffect(() => {
		const onResize = () => {
			setColumns(previousColumns => {
				if(window.innerWidth < 1200){
					if(previousColumns.length !== 3){
						previousColumns.length = 3;
						return previousColumns.concat([]);
					}
				}
				else{
					if(previousColumns.length === 3){
						return previousColumns.concat([{
							name: 'description',
							header: () => (
								<div className={'text-center innerTdHead'}>
									<FormattedMessage {...taskTableMsg.taskdesc} />
								</div>
							),
							cell: (content: any) => {
								return innerTaskCell(content.data, <span>{content.data.shortenDescription}</span>);
							},
							sortable: false
						}]);
					}
				}

				return previousColumns;
			});
		};

		onResize();
		window.addEventListener("resize", onResize);
		return () => window.removeEventListener("resize", onResize);
	}, [innerTaskCell]);

	return (
		<Fragment>
			<div className={classNames}>
				<span><FontAwesomeIcon icon={faTasks} />{' '}<b><FormattedMessage {...taskTableMsg.tasklistTitle} /></b>
					{equipment && <Button color="light" size="sm" className="float-right mb-2" onClick={() => modalHook.displayData(createDefaultTask(equipment)) } aria-label="Add"><FontAwesomeIcon icon={faPlusSquare} /></Button>}
				</span>
				{areTasksLoading ? <Loading/> :
				<Table
					data={getTasksData()}
					className="default-theme"
					defaultSortParameter="status"
					defaultSortDirection="desc"
					columns={columns}
				/>}
			</div>
			{equipment !== undefined && modalHook.data !== undefined && <ModalEditTask  equipment={equipment}
							task={modalHook.data}
							onTaskSaved={onTaskSaved} 
							visible={modalHook.editModalVisibility} 
							toggle={modalHook.toggleModal}
							className='modal-dialog-centered'/>}
		</Fragment>
	);
}

export default React.memo(TaskTable);