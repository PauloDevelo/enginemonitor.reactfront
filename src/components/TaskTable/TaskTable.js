import React from 'react';
import { Table, Button } from 'reactstrap';
import { FormattedMessage } from 'react-intl';
import { faTasks, faPlusSquare } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { CSSTransition, TransitionGroup } from 'react-transition-group'
import PropTypes from 'prop-types';

import TaskRow from './TaskRow';

import taskTableMsg from "./TaskTable.messages";

export const TaskTable = ({tasks, changeCurrentTask, toggleModal, classNames}) => {
	var listLines = [];
	if(tasks){
		const trStyle = { cursor: 'pointer' };
		listLines = tasks.map((task) => {
			return(
				<CSSTransition key={task._id} in={true} timeout={500} classNames="tr">
					<TaskRow task={task} trStyle={trStyle} onClick={() => changeCurrentTask(task) } />
				</CSSTransition>
			);
		});
	}

	return (
		<div className={classNames}>
			<span className="mb-2"><FontAwesomeIcon icon={faTasks} />{' '}<b><FormattedMessage {...taskTableMsg.tasklistTitle} /></b>
			<Button color="light" size="sm" className="float-right mb-2" onClick={() => toggleModal() }><FontAwesomeIcon icon={faPlusSquare} /></Button></span>
			<Table responsive size="sm" hover>
				<thead className="thead-light">
					<tr>
						<th><FormattedMessage {...taskTableMsg.taskname} /></th>
						<th><FormattedMessage {...taskTableMsg.todo} /></th>
						<th><FormattedMessage {...taskTableMsg.taskdesc} /></th>
					</tr>
				</thead>
				<TransitionGroup component="tbody">
					{listLines}
				</TransitionGroup>
			</Table>
		</div>
	);
}

TaskTable.propTypes = {
	tasks: PropTypes.array.isRequired,
	changeCurrentTask: PropTypes.func.isRequired,
	toggleModal: PropTypes.func.isRequired,
	classNames: PropTypes.string
};

export default TaskTable;