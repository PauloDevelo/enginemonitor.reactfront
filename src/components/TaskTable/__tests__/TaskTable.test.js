import chai, { assert } from 'chai';

import React from 'react';
import { IntlProvider } from 'react-intl';

import { mount } from 'enzyme';

// eslint-disable-next-line no-unused-vars
import localforage from 'localforage';
import ignoredMessages from '../../../testHelpers/MockConsole';

import taskProxy from '../../../services/TaskProxy';

import { AgeAcquisitionType } from '../../../types/Types';
import TaskTable from '../TaskTable';
import updateWrapper from '../../../testHelpers/EnzymeHelper';

jest.mock('localforage');
jest.mock('../../../services/TaskProxy');

chai.use(require('chai-datetime'));

describe('TaskTable', () => {
  function getSortedTasks(taskArray) {
    const copyTaskArray = taskArray.concat([]);
    return copyTaskArray.sort((a, b) => {
      if (a.nextDueDate > b.nextDueDate) {
        return -1;
      }
      if (a.nextDueDate < b.nextDueDate) {
        return 1;
      }
      return 0;
    });
  }

  const equipment = {
    _uiId: 'equipment_01',
    name: 'moteur',
    brand: 'Nanni',
    model: 'N3.30',
    age: 2750,
    installation: new Date('2019-11-07T23:39:36.288Z'),
    ageAcquisitionType: AgeAcquisitionType.manualEntry,
    ageUrl: '',
  };

  const tasks = [{
    _uiId: 'task_01',
    name: 'vidange',
    usagePeriodInHour: 200,
    periodInMonth: 6,
    description: 'Change the oil',
    nextDueDate: new Date('2019-11-07T23:39:36.288Z'),
    level: 0,
    usageInHourLeft: 100,
  },
  {
    _uiId: 'task_02',
    name: 'change filtre a huile',
    usagePeriodInHour: 400,
    periodInMonth: 12,
    description: 'Change le filtre a huile',
    nextDueDate: new Date('2019-12-07T23:39:36.288Z'),
    level: 0,
    usageInHourLeft: 300,
  },
  {
    _uiId: 'task_03',
    name: "change l'impeller",
    usagePeriodInHour: 800,
    periodInMonth: 24,
    description: "Change l'impeller",
    nextDueDate: new Date('2020-11-08T23:39:36.288Z'),
    level: 0,
    usageInHourLeft: 600,
  },
  {
    _uiId: 'task_04',
    name: 'change la courroie',
    usagePeriodInHour: 1000,
    periodInMonth: 24,
    description: 'Change la courroie de distribution',
    nextDueDate: new Date('2020-11-09T23:39:36.288Z'),
    level: 0,
    usageInHourLeft: 800,
  }];

  beforeAll(() => {
    ignoredMessages.length = 0;
    ignoredMessages.push('test was not wrapped in act(...)');
    ignoredMessages.push('[React Intl] Missing message');
  });

  beforeEach(() => {
  });

  afterEach(() => {
    taskProxy.existTask.mockReset();
  });

  it('Should render render the loading spinner while loading the tasks', async () => {
    // Arrange
    const onTaskSaved = jest.fn();
    const changeCurrentTask = jest.fn();

    // Act
    const taskTable = mount(
      <IntlProvider locale={navigator.language}>
        <TaskTable
          equipment={equipment}
          tasks={[]}
          areTasksLoading
          onTaskSaved={onTaskSaved}
          changeCurrentTask={changeCurrentTask}
        />
      </IntlProvider>,
    );

    // Assert
    expect(taskTable.find('tbody').length).toBe(0);
    expect(taskTable.find('Loading').length).toBe(1);

    expect(onTaskSaved).toBeCalledTimes(0);
    expect(changeCurrentTask).toBeCalledTimes(0);

    expect(taskTable).toMatchSnapshot();
  });

  it('Should render all the tasks sorted by due date', async () => {
    // Arrange
    const onTaskSaved = jest.fn();
    const changeCurrentTask = jest.fn();

    // Act
    const taskTable = mount(
      <IntlProvider locale={navigator.language}>
        <TaskTable
          equipment={equipment}
          tasks={tasks}
          areTasksLoading={false}
          onTaskSaved={onTaskSaved}
          changeCurrentTask={changeCurrentTask}
        />
      </IntlProvider>,
    );
    await updateWrapper(taskTable);

    // Assert
    const sortedTasks = getSortedTasks(tasks);
    const tbodyProps = taskTable.find('tbody').at(0).props();
    sortedTasks.forEach((task, index) => {
      expect(tbodyProps.children[index].props.data.task).toBe(task);
    });

    expect(onTaskSaved).toBeCalledTimes(0);
    expect(changeCurrentTask).toBeCalledTimes(0);
  });


  it('Should call changeCurrentTask when clicking on any cell', async () => {
    // Arrange
    const onTaskSaved = jest.fn();
    const changeCurrentTask = jest.fn();

    const taskTable = mount(
      <IntlProvider locale={navigator.language}>
        <TaskTable
          equipment={equipment}
          tasks={tasks}
          areTasksLoading={false}
          onTaskSaved={onTaskSaved}
          changeCurrentTask={changeCurrentTask}
        />
      </IntlProvider>,
    );
    await updateWrapper(taskTable);

    const cells = taskTable.find('ClickableCell');

    const sortedTasks = getSortedTasks(tasks);
    let clickCounter = 0;
    for (let index = 0; index < sortedTasks.length; index++) {
      const task = sortedTasks[index];
      for (let numColumn = 0; numColumn < 3; numColumn++) {
        // Act
        cells.at(index * 3 + numColumn).simulate('click');
        // eslint-disable-next-line no-await-in-loop
        await updateWrapper(taskTable);

        // Assert
        expect(changeCurrentTask).toHaveBeenCalledTimes(clickCounter + 1);
        expect(changeCurrentTask.mock.calls[clickCounter][0]).toBe(task);

        clickCounter++;
      }
    }
  });

  it('should display the task edition modal to add a new entry', async () => {
    // Arrange
    const onTaskSaved = jest.fn();
    const changeCurrentTask = jest.fn();

    taskProxy.existTask.mockImplementation(async () => Promise.resolve(false));

    const taskTable = mount(
      <IntlProvider locale={navigator.language}>
        <TaskTable
          equipment={equipment}
          tasks={tasks}
          areTasksLoading={false}
          onTaskSaved={onTaskSaved}
          changeCurrentTask={changeCurrentTask}
        />
      </IntlProvider>,
    );
    await updateWrapper(taskTable);

    const createButton = taskTable.find('Button');

    // Act
    createButton.simulate('click');
    await updateWrapper(taskTable);

    // Assert
    const editTaskModal = taskTable.find('ModalEditTask');
    expect(editTaskModal.props().visible).toBe(true);
    expect(editTaskModal.props().onTaskSaved).toBe(onTaskSaved);
    expect(onTaskSaved).toBeCalledTimes(0);
  });

  // it('Should add an entry and it should remain sorted by date', async () => {
  //   // Arrange
  //   const onHistoryChanged = jest.fn();

  //   const historyTaskTable = mount(
  //     <IntlProvider locale={navigator.language}>
  //       <HistoryTaskTable
  //         equipment={equipment}
  //         task={task}
  //         onHistoryChanged={onHistoryChanged}
  //         taskHistoryRefreshId={0}
  //       />
  //     </IntlProvider>,
  //   );
  //   await updateWrapper(historyTaskTable);

  //   const addButton = historyTaskTable.find('Button');
  //   addButton.simulate('click');
  //   await updateWrapper(historyTaskTable);

  //   const editEntryModal = historyTaskTable.find('ModalEditEntry');
  //   const { saveEntry } = editEntryModal.props();

  //   const newEntry = {
  //     _uiId: 'entry_088',
  //     name: 'remplacement silent bloc',
  //     date: new Date('2019-09-08T00:11:18.112Z'),
  //     age: 125894,
  //     remarks: 'RAS',
  //     taskUiId: task._uiId,
  //     equipmentUiId: equipment._uiId,
  //   };
  //   // Act
  //   saveEntry(newEntry);
  //   await updateWrapper(historyTaskTable);

  //   // Assert
  //   expect(editEntryModal.length).toBe(1);
  //   expect(editEntryModal.props().visible).toBe(true);
  //   expect(editEntryModal.props().equipment).toBe(equipment);

  //   expect(historyTaskTable.find('ClickableCell').length).toBe((entries.length + 1) * 3);
  //   const newCells = historyTaskTable.find('ClickableCell').findWhere((n) => n.props().data === newEntry);
  //   expect(newCells.length).toBe(3);

  //   assertTableSortedByDate(historyTaskTable);
  //   expect(onHistoryChanged).toBeCalledTimes(1);
  // });

  // it('Should remove an entry', async () => {
  //   // Arrange
  //   const onHistoryChanged = jest.fn();

  //   const historyTaskTable = mount(
  //     <IntlProvider locale={navigator.language}>
  //       <HistoryTaskTable
  //         equipment={equipment}
  //         task={task}
  //         onHistoryChanged={onHistoryChanged}
  //         taskHistoryRefreshId={0}
  //       />
  //     </IntlProvider>,
  //   );
  //   await updateWrapper(historyTaskTable);

  //   const cells = historyTaskTable.find('ClickableCell').at(5);
  //   cells.simulate('click');
  //   const entryToDelete = cells.props().data;
  //   await updateWrapper(historyTaskTable);

  //   const editEntryModal = historyTaskTable.find('ModalEditEntry');
  //   const { deleteEntry } = editEntryModal.props();

  //   // Act
  //   deleteEntry(entryToDelete);
  //   await updateWrapper(historyTaskTable);

  //   // Assert
  //   expect(editEntryModal.length).toBe(1);
  //   expect(editEntryModal.props().visible).toBe(true);
  //   expect(editEntryModal.props().equipment).toBe(equipment);
  //   expect(editEntryModal.props().entry).toBe(entryToDelete);

  //   expect(historyTaskTable.find('ClickableCell').length).toBe((entries.length - 1) * 3);
  //   const newCells = historyTaskTable.find('ClickableCell').findWhere((n) => n.props().data === entryToDelete);
  //   expect(newCells.length).toBe(0);

  //   expect(onHistoryChanged).toBeCalledTimes(1);
  // });
});
