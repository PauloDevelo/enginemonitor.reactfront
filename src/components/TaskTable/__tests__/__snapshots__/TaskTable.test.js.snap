// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskTable Should render render the loading spinner while loading the tasks 1`] = `
<IntlProvider
  defaultFormats={Object {}}
  defaultLocale="en"
  formats={Object {}}
  locale="en-US"
  messages={Object {}}
  onError={[Function]}
  textComponent={Symbol(react.fragment)}
>
  <Memo(TaskTable)
    areTasksLoading={true}
    changeCurrentTask={[MockFunction]}
    equipment={
      Object {
        "_uiId": "equipment_01",
        "age": 2750,
        "ageAcquisitionType": 1,
        "ageUrl": "",
        "brand": "Nanni",
        "installation": 2019-11-07T23:39:36.288Z,
        "model": "N3.30",
        "name": "moteur",
      }
    }
    onTaskSaved={[MockFunction]}
    tasks={Array []}
  >
    <div
      className="tasktable"
    >
      <span>
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f0ae",
                "M139.61 35.5a12 12 0 0 0-17 0L58.93 98.81l-22.7-22.12a12 12 0 0 0-17 0L3.53 92.41a12 12 0 0 0 0 17l47.59 47.4a12.78 12.78 0 0 0 17.61 0l15.59-15.62L156.52 69a12.09 12.09 0 0 0 .09-17zm0 159.19a12 12 0 0 0-17 0l-63.68 63.72-22.7-22.1a12 12 0 0 0-17 0L3.53 252a12 12 0 0 0 0 17L51 316.5a12.77 12.77 0 0 0 17.6 0l15.7-15.69 72.2-72.22a12 12 0 0 0 .09-16.9zM64 368c-26.49 0-48.59 21.5-48.59 48S37.53 464 64 464a48 48 0 0 0 0-96zm432 16H208a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h288a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-320H208a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h288a16 16 0 0 0 16-16V80a16 16 0 0 0-16-16zm0 160H208a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h288a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16z",
              ],
              "iconName": "tasks",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-tasks fa-w-16 "
            data-icon="tasks"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M139.61 35.5a12 12 0 0 0-17 0L58.93 98.81l-22.7-22.12a12 12 0 0 0-17 0L3.53 92.41a12 12 0 0 0 0 17l47.59 47.4a12.78 12.78 0 0 0 17.61 0l15.59-15.62L156.52 69a12.09 12.09 0 0 0 .09-17zm0 159.19a12 12 0 0 0-17 0l-63.68 63.72-22.7-22.1a12 12 0 0 0-17 0L3.53 252a12 12 0 0 0 0 17L51 316.5a12.77 12.77 0 0 0 17.6 0l15.7-15.69 72.2-72.22a12 12 0 0 0 .09-16.9zM64 368c-26.49 0-48.59 21.5-48.59 48S37.53 464 64 464a48 48 0 0 0 0-96zm432 16H208a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h288a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-320H208a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h288a16 16 0 0 0 16-16V80a16 16 0 0 0-16-16zm0 160H208a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h288a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </FontAwesomeIcon>
         
        <b>
          <FormattedMessage
            defaultMessage="Task list"
            description="Panel title"
            id="tasklistTitle"
            values={Object {}}
          >
            Task list
          </FormattedMessage>
        </b>
        <Button
          aria-label="Add"
          className="float-right mb-2"
          color="light"
          onClick={[Function]}
          size="sm"
          tag="button"
        >
          <button
            aria-label="Add"
            className="float-right mb-2 btn btn-light btn-sm"
            onClick={[Function]}
            type="button"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    448,
                    512,
                    Array [],
                    "f0fe",
                    "M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-32 252c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92H92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z",
                  ],
                  "iconName": "plus-square",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              swapOpacity={false}
              symbol={false}
              title=""
              transform={null}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-plus-square fa-w-14 "
                data-icon="plus-square"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-32 252c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92H92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </FontAwesomeIcon>
          </button>
        </Button>
      </span>
      <Loading>
        <div
          className="lds-spinner"
        >
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
        </div>
      </Loading>
    </div>
  </Memo(TaskTable)>
</IntlProvider>
`;
